app/controllers/application_controller.rb

```diff
@@ -22,6 +22,26 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  #By default only show breadcrumbs to logged-in users.
+  def show_breadcrumbs
+    current_ability.current_user.logged_in?
+  end
+  helper_method :show_breadcrumbs
+
+  # This is used in generic_works_controller and collection_show_controller.
+  # It resolves issue https://github.com/sciencehistory/chf-sufia/issues/885
+  def special_breadcrumb_filter (the_breadcrumbs)
+    return if the_breadcrumbs == nil
+    # If the user's logged in, leave as is.
+    return if current_ability.current_user.logged_in?
+    # Otherwise, only show the "Back to Search Results" breadcrumb,
+    # and suppress all other breadcrumbs. ( This method modifies
+    # the_breadcrumbs in place. )
+    back_str = I18n.t('sufia.bread_crumb.search_results')
```

Comment:
comment "this method modifies the breadcrumbs in place" should go in comment above method definition, inside of inside.  Since it's important. 

I'm being extra detailed just as a learning exersize here, some of these are admittedly nitpicks!

Reply:
Really appreciate your taking the time- very helpful!